src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -310,27 +311,27 @@ protected void saveAnUpdateConfigs(final Client client, final RestRequest reques
 							.setIfSeqNo(configuration.getSeqNo())
 							.setIfPrimaryTerm(configuration.getPrimaryTerm())
 							.source(id, XContentHelper.toXContent(configuration, XContentType.JSON, false)),
-					new ConfigUpdatingActionListener<IndexResponse>(client, actionListener));
+					new ConfigUpdatingActionListener<>(new String[]{id}, client, actionListener));
 		} catch (IOException e) {
 			throw ExceptionsHelper.convertToElastic(e);
 		}
 	}
 
-	protected static class ConfigUpdatingActionListener<Response> implements ActionListener<Response>{
-
+	protected static class ConfigUpdatingActionListener<Response> implements ActionListener<Response> {
+		private final String[] cTypes;
 		private final Client client;
 		private final ActionListener<Response> delegate;
 
-		public ConfigUpdatingActionListener(Client client, ActionListener<Response> delegate) {
-			super();
-			this.client = client;
-			this.delegate = delegate;
+		public ConfigUpdatingActionListener(String[] cTypes, Client client, ActionListener<Response> delegate) {
+			this.cTypes = Objects.requireNonNull(cTypes, "cTypes must not be null");
```

Comment:
Should we validate that the list of strings contains valid cTypes ? What happens if it will get some invalid CType ?

Reply:
The validation is done on the target node. Note that even if we validate that `cType` is valid on the source node, it may still be invalid on the target node.